### PR TITLE
[MI-2332]:Create /confluence migrate cleanup to delete all old subscriptions

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**

### CODE REVIEW

File: server/command.go [hunk 1]
```diff
@@ -51,36 +52,40 @@ const (
...
+	invalidCommand             = "Invalid command."
+	installOnlySystemAdmin     = "`/confluence install` can only be run by a system administrator."
+	configNotFoundError        = "configuration not found for %s. Please ask system admin to add config for %s in plugin configuration"
+	configServerURL            = "Server URL"
+	configClientID             = "Client ID"
+	configClientSecret         = "Client Secret"
+	configAPIEndpoint          = "%s/api/v4/actions/dialogs/open"
+	configModalTitle           = "Confluence Config"
+	configPerPage              = 10
+	NoOldSubscriptionMsg       = "No old Subscriptions found for migration"
```
Comment (raghavaggarwal2308) [inline server/command.go]:
```suggestion
	NoOldSubscriptionsMsg       = "No old Subscriptions found for migration."
```

File: server/command.go [hunk 2]
```diff
@@ -51,36 +52,40 @@ const (
...
+	NoOldSubscriptionDeleteMsg = "No old Subscriptions were found for cleanup."
```
Comment (raghavaggarwal2308) [inline server/command.go]:
```suggestion
	NoOldSubscriptionsDeleteMsg = "No old Subscriptions were found for cleanup."
```

**Review by raghavaggarwal2308 — APPROVED** [review]

File: server/command.go [hunk 3]
```diff
@@ -51,36 +52,40 @@ const (
...
+		"install/cloud":   showInstallCloudHelp,
+		"install/server":  showInstallServerHelp,
+		"uninstall":       executeInstanceUninstall,
+		"help":            confluenceHelpCommand,
+		"config/add":      addConfig,
+		"config/list":     listConfig,
+		"config/delete":   deleteConfig,
+		"migrate/list":    listOldSubscriptions,
+		"migrate/start":   startSubscriptionMigration,
+		"migrate/cleanup": deleteOldSubscription,
```
Comment (ayusht2810) [inline server/command.go]:
```suggestion
		"migrate/cleanup": deleteOldSubscriptions,
```

File: server/command.go [hunk 4]
```diff
@@ -466,6 +484,35 @@ func startSubscriptionMigration(p *Plugin, context *model.CommandArgs, args ...s
...
+
+	if len(oldSubscriptions) == 0 {
+		p.postCommandResponse(context, NoOldSubscriptionDeleteMsg)
+		return &model.CommandResponse{}
+	}
+
+	go func() {
+		if err := p.API.KVDelete(storePackage.GetOldSubscriptionKey()); err != nil {
+			p.API.LogError("Unable to delete old subscriptions", "Error", err.Error())
+		}
```
Comment (ayusht2810) [inline server/command.go:507]: shouldn't we return from here?

Comment (Kshitij-Katiyar) [reply server/command.go:507]: no no in the error case we are just loggin the error in server logs and all the processes will be there as it is

**Review by ayusht2810 — APPROVED** [review]

File: server/command.go [hunk 5]
```diff
@@ -51,36 +52,40 @@ const (
...
+	installOnlySystemAdmin      = "`/confluence install` can only be run by a system administrator."
+	configNotFoundError         = "configuration not found for %s. Please ask system admin to add config for %s in plugin configuration"
+	configServerURL             = "Server URL"
+	configClientID              = "Client ID"
+	configClientSecret          = "Client Secret"
+	configAPIEndpoint           = "%s/api/v4/actions/dialogs/open"
+	configModalTitle            = "Confluence Config"
+	configPerPage               = 10
+	NoOldSubscriptionsMsg       = "No old Subscriptions found for migration"
+	NoOldSubscriptionsDeleteMsg = "No old Subscriptions were found for cleanup."
```
Comment (manojmalik20) [inline server/command.go]:
```suggestion
	NoOldSubscriptionsMsg       = "No old subscriptions found for migration."
	NoOldSubscriptionsDeleteMsg = "No old subscriptions were found for cleanup."
```

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**